patches/build-config-compiler-BUILD.gn.patch

```diff
@@ -0,0 +1,154 @@
++  } else if (is_ios) {
++    import("//build/config/ios/config.gni")  # For `target_environment`
++
++    _dir = "darwin"
++    if (target_environment == "simulator") {
++      _libname = "iossim"
++    } else {
++      _libname = "ios"
++    }
++  } else if (is_linux || is_chromeos) {
++    if (current_cpu == "x64") {
++      _dir = "x86_64-unknown-linux-gnu"
++    } else if (current_cpu == "x86") {
++      _dir = "i386-unknown-linux-gnu"
++    } else if (current_cpu == "arm") {
++      _dir = "armv7-unknown-linux-gnueabihf"
++    } else {
++      assert(false)  # Unhandled cpu type
++    }
++  } else if (is_fuchsia) {
++    if (current_cpu == "x64") {
++      _dir = "x86_64-unknown-fuchsia"
++    } else if (current_cpu == "arm64") {
++      _dir = "aarch64-unknown-fuchsia"
++    } else {
++      assert(false)  # Unhandled cpu type
++    }
++  } else if (is_android) {
++    _dir = "linux"
++    if (current_cpu == "x64") {
++      _suffix = "-x86_64-android"
++    } else if (current_cpu == "x86") {
++      _suffix = "-i686-android"
++    } else if (current_cpu == "arm") {
++      _suffix = "-arm-android"
++    } else if (current_cpu == "arm64") {
++      _suffix = "-aarch64-android"
++    } else {
++      assert(false)  # Unhandled cpu type
++    }
++  } else {
++    assert(!toolchain_has_rust)  # Unhandled target platform
++  }
++
++  _clang_lib_dir = "$clang_base_path/lib/clang/$clang_version/lib"
++  _lib_file = "${_prefix}clang_rt.${_libname}${_suffix}.${_ext}"
++  libs = [ "$_clang_lib_dir/$_dir/$_lib_file" ]
++}
++
+ # runtime_library -------------------------------------------------------------
+ #
+ # Sets the runtime library and associated options.
+@@ -1575,6 +1659,10 @@ config("runtime_library") {
+     configs += [ "//build/config/c++:runtime_library" ]
+   }
+ 
++  if (toolchain_has_rust && target_os != "win" && target_os != "arm64") {
++    configs += [ ":compiler_builtins" ]
++  }
++
+   # TODO(crbug.com/830987): Come up with a better name for is POSIX + Fuchsia
+   # configuration.
+   if (is_posix || is_fuchsia) {
+@@ -1746,6 +1834,11 @@ config("default_warnings") {
+           # TODO(crbug.com/1412713) Evaluate and possibly enable.
+           "-Wno-deprecated-this-capture",
+         ]
++
++        if (llvm_force_head_revision) {
++          # TODO(https://crbug.com/1448905): Evaluate and possibly enable.
++          cflags += [ "-Wno-builtin-macro-redefined" ]
++        }
+       }
+     }
+   }
+@@ -1951,7 +2044,7 @@ config("no_chromium_code") {
+ config("noshadowing") {
+   # This flag has to be disabled for nacl because the nacl compiler is too
+   # strict about shadowing.
+-  if (is_clang && (!is_nacl || is_nacl_saigo)) {
++  if (is_clang && (!is_nacl || is_nacl_saigo) && !llvm_force_head_revision) {
+     cflags = [ "-Wshadow" ]
+   }
+ }
+@@ -2534,7 +2627,7 @@ config("symbols") {
+       if (use_dwarf5) {
+         cflags += [ "-gdwarf-5" ]
+         rustflags += [ "-Zdwarf-version=5" ]
+-      } else if (!is_apple) {
```

Comment:
this is not an upstream patch, but I think it will need to be eventually or they'll get the same error I'm seeing for different dwarf versions